### PR TITLE
test-drove schema updates and refactored some old-style code

### PR DESCRIPTION
**Pull request by adamwoodcock98**
## Before you start
1. Is there an issue on the repository for this request? If not, please create one.
2. Have you followed the correct workflow in the [team documentation](https://docs.google.com/document/d/13yi709ax4nmfTG9J0F57yEFpYRH7fdtczwvVJCrNIsM/edit#)?
3. Conflict resolution to be dealt with **on GitHub**.

## What do these code changes do?
Added `user` to the posts schema, test-driven. Updated `post` model to modernise code and test requirements of user and message keys.

Removed `New:` from the `PostsController` object.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have created a PR ticket on the [Trello board](https://trello.com/b/P8lEsvZe/acebook).
- [x] I will link the PR to the Trello card once created.
- [x] I will send the PR in the Slack channel and ping @here.

### CODE REVIEW

**Comment by adamwoodcock98** [comment]
[test-drove schema updates and refactored some old-style code #28](https://trello.com/c/DVmN2BuC/41-test-drove-schema-updates-and-refactored-some-old-style-code-28)

File: spec/models/post.spec.js [hunk 1]
```diff
@@ -24,15 +47,15 @@ describe("Post model", () => {
   });
 
   it("can save a post", (done) => {
-    var post = new Post({ message: "some message" });
+    var post = new Post({ message: "can save a post", user: mockUserId });
```
Comment (PKilgarriff) [inline spec/models/post.spec.js:50]: You've updated most of the references from var to const - would be worth catching this one too

**Review by PKilgarriff — CHANGES_REQUESTED** [review]
Overall fine - spec/models/post.spec now has inconsistency between var and const - one suggested update

**Review by clovellbsc — APPROVED** [review]

File: spec/models/post.spec.js (hunk 1 above)
Comment (adamwoodcock98) [reply spec/models/post.spec.js]: Resolved in newest commit

**Review by PKilgarriff — APPROVED** [review]
All good, cheers!

**Pull request merged**